tests/unit/test_cmr_search.py

```diff
@@ -238,4 +256,78 @@ def test_exceptions_granule_link(self):
         with self.subTest('Granule has `links` and all correct attributes'):
             self.assertEqual(get_granule_link(granule_response_links_correct),
                              'https://data.gesdisc.earthdata.nasa.gov/.nc4')
-    
\ No newline at end of file
+    
+    
+    def test_download_gran_exceptions(self):
+        ''' Check if exceptions are raised for incorrect input parameters
+        '''
+    
+        with self.assertRaises(MissingPositionalArguments):
+            # Input parameter, token, is missing
+            download_granule(granule_link='https://foo.gov/example.nc4')
+            
+        with self.assertRaises(NotADirectoryError):
+            # Input parameter, `out_directory` should be a real directory
+            download_granule('https://foo.gov/example.nc4', 
+                             '/Usr/foo/dir',
+                             'foo')
+        
+        with self.assertRaises(MissingPositionalArguments):
+            # Missing argument, `granule_link`
+            download_granule(token='foo')
+        
+        with self.assertRaises(ValueError):
+            # `granule_link` should be str or bytes like not int
+            download_granule(granule_link=0, token='foo')
+        
+        with self.assertRaises(GranuleDownloadException):
+            # Not a real https link
+            download_granule('https://foo.gov/example.nc4', token='foo')
+    
+    
+    def _mock_requests(self, status=200, content="CONTENT"):
+        '''
+        Mock requests.get module and it's methods content and status.
+        Return the mock object.
+        '''
+        mock_response = Mock(spec=requests.Response)
+        mock_response.status_code = status
+        mock_response.content = content
+        return mock_response
+
+
+    @patch('requests.get')
+    def test_download_granule(self, mock_requests_get):
+        ''' Check if `download_granules` returns the expected
+            content for the mocked response.
+        '''
+        link = 'https://foo.gov/example.nc4'
+        # Set mock_content as bytes for writing
+        mock_content = bytes(link, encoding='utf-8')
+        # Set the mock_response with the `_mock_requests` object content method
+        mock_response = self._mock_requests(content=mock_content)
+        # Set the return_value of `mock_requests_get` to mock_response
+        mock_requests_get.return_value = mock_response
+        file_path = download_granule(link, self.output_dir, token='foo')
+        # Check if `download_granule` was called once with expected parameters
+        mock_requests_get.assert_called_once_with(link, headers=
+                                                  {'Authorization':
+                                                      f'Bearer {"foo"}'},
+                                                  timeout=10)
```

Comment:
Nit: There's a little bit of wonky indentation here. Maybe:

```
        mock_requests_get.assert_called_once_with(
            link,
            headers={'Authorization': f'Bearer {"foo"}'},
            timeout=10
        )
```

Second nit: You don't really need the f-string in your authorisation header value, if you are just explicitly naming a string within the string. (I said string too many times in a row there...)